metaissue.go

```diff
@@ -58,10 +70,65 @@ func (s *IssueService) GetCreateMeta(projectkeys string) (*CreateMetaInfo, *Resp
 	return s.GetCreateMetaWithContext(context.Background(), projectkeys)
 }
 
+// GetCreateMetaWithOptionsWithContextForJira9 makes the Jira 9 api call to get the meta information without requiring to have a projectKey
+func (s *IssueService) GetCreateMetaWithOptionsWithContextForJira9(ctx context.Context, options *GetQueryOptions) (*CreateMetaInfo, *Response, error) {
+	apiEndpoint := fmt.Sprintf("rest/api/2/project/%s", options.ProjectKeys)
+
+	req, err := s.client.NewRequestWithContext(ctx, "GET", apiEndpoint, nil)
+	if err != nil {
+		return nil, nil, fmt.Errorf("creating request to %s got error: %v", apiEndpoint, err)
+	}
+	metaProject := new(MetaProject)
+	resp, err := s.client.Do(req, metaProject)
+	if err != nil {
+		return nil, resp, fmt.Errorf("error from %q: %v", apiEndpoint, err)
+	}
+
+	for i, issueType := range metaProject.IssueTypes {
+		issueTypeApiEndpoint := fmt.Sprintf("rest/api/2/issue/createmeta/%s/issuetypes/%s", options.ProjectKeys, issueType.Id)
+		req, err := s.client.NewRequestWithContext(ctx, "GET", issueTypeApiEndpoint, nil)
+		if err != nil {
+			return nil, nil, fmt.Errorf("creating request to %s got error: %v", issueTypeApiEndpoint, err)
+		}
+		details := new(MetaIssueTypeDetails)
+		if _, err := s.client.Do(req, details); err != nil {
+			return nil, nil, fmt.Errorf("error from %q: %v", issueTypeApiEndpoint, err)
+		}
+		metaProject.IssueTypes[i].Fields = tcontainer.MarshalMap{}
+		for _, v := range details.Values {
+			fieldId, err := v.String("fieldId")
+			if err != nil {
+				continue
+			}
+			metaProject.IssueTypes[i].Fields[fieldId] = v
+		}
+	}
+	meta := &CreateMetaInfo{Projects: []*MetaProject{metaProject}}
+	return meta, resp, nil
+}
+
+// todo: we need to log the errors here
+func (s *IssueService) GetJiraAPIVersion(ctx context.Context) int {
+	apiEndpoint := "rest/api/2/serverInfo"
+	req, err := s.client.NewRequestWithContext(ctx, "GET", apiEndpoint, nil)
+	if err != nil {
+		return 0
+	}
+	version := new(JiraServerInfo)
+
+	_, err = s.client.Do(req, version)
+	if err != nil {
+		return 0
+	}
+	return version.VersionNumbers[0]
```

Comment:
What else is in this slice? Could we document it here maybe?

Reply:
add comments.